cloudflare/resource_cloudflare_worker_kv_test.go

```diff
@@ -0,0 +1,91 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccCloudflareWorkersKV_Basic(t *testing.T) {
+	t.Parallel()
+	var kvPair cloudflare.WorkersKVPair
+	name := generateRandomResourceName()
+	key := generateRandomResourceName()
+	value := generateRandomResourceName()
+	resourceName := "cloudflare_workers_kv." + name
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCloudflareWorkersKVDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareWorkersKV(name, key, value),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWorkersKVExists(key, &kvPair),
+					resource.TestCheckResourceAttr(
+						resourceName, "value", value,
+					),
+				),
+			},
+		},
+	})
+}
+
+func testAccCloudflareWorkersKVDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*cloudflare.API)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cloudflare_workers_kv" {
+			continue
+		}
+
+		namespaceID := rs.Primary.Attributes["namespace_id"]
+		key := rs.Primary.Attributes["key"]
+
+		_, err := client.ReadWorkersKV(context.Background(), namespaceID, key)
+
+		if err == nil {
+			return fmt.Errorf("workers kv pair still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckCloudflareWorkersKV(rName string, key string, value string) string {
+	return testAccCheckCloudflareWorkersKVNamespace(rName) + fmt.Sprintf(`
+resource "cloudflare_workers_kv" "%[1]s" {
+	namespace_id = "${cloudflare_workers_kv_namespace.%[1]s.id}"
```

Comment:
This test isn't going to be 0.11 and 0.12 compatible due to the change in the way variables are referenced. We have a couple of options here:

- Only add test cases for 0.12.x (as that is what we're running in CI); or
- Don't use variables and instead explicitly reference the resources with hardcoded strings.